File: calls.py
from dotenv import load_dotenv

import requests
import json
import os

load_dotenv()

url = os.environ['domain']
secret = os.environ['secret']

headers = {
    'Content-Type': 'application/json',
    'Authorization': 'Bearer ' + secret
}


def updateStatus(tid, progress=None, duration=None, paused=None):
    payload = json.dumps({
        "tid": tid,
        "progress": progress,
        "paused": paused,
        "duration": duration
    })

    response = requests.request(
        "PUT", url + "/player/set/status", headers=headers, data=payload)
    return response.json()


def updateNext(tid):
    payload = json.dumps({
        "tid": tid
    })

    response = requests.request(
        "PUT", url + "/player/set/next", headers=headers, data=payload)
    return response.json()


def getTop():
    response = requests.request(
        "GET", url + "/player/get/top", headers=headers)
    return response.json()

File: src/modules/calls.js
import { domain, secret, verbose } from '../../env.js'

import got from 'got'

import { logAction, logValue } from './log.js'

const headers = { authorization: 'Bearer ' + secret }

function updateStatus(track, progress = null, duration = null, paused) {
	return new Promise((resolve, reject) => {
		console.log({
			tid: track?.tid ?? null,
			progress,
			duration,
			paused: paused ?? track ? false : true,
		})
		got
			.put(`https://${domain}/api/player/set/status`, {
				headers,
				json: {
					tid: track?.tid ?? null,
					progress,
					duration,
					paused: paused ?? track ? false : true,
				},
			})
			.json()
			.then((ok) => resolve(ok))
			.catch((err) => reject(err))

		verbose > 1 &&
			console.info(`PUT status`, track?.tid, progress, duration, paused)
	})
}

function updateNext(track) {
	return new Promise((resolve, reject) => {
		got
			.put(`https://${domain}/api/player/set/next`, {
				headers,
				json: {
					tid: track ? track.tid : null,
				},
			})
			.json()
			.then((ok) => resolve(ok))
			.catch((err) => reject(err))

		verbose > 1 && console.info(`PUT next`, track?.tid)
	})
}

function getTop(mode = 'once') {
	logAction(`Fetching tids`, mode, domain)

	return got
		.get(`https://${domain}/api/player/get/top?mode=${mode}`, {
			headers,
		})
		.json()
		.then((tids) => {
			logValue('Tracks', tids.length)
			return tids
		})
}

function getTrack(tid) {
	return got.get(`https://${domain}/api/track/${tid}`).json()
}

export { updateStatus, updateNext, getTop, getTrack }
